From 8a5bfcdc6b12f84addb4cb582414794e65c1fb4b Mon Sep 17 00:00:00 2001
From: Derek Weitzel <djw8605@gmail.com>
Date: Wed, 20 Sep 2017 10:18:34 -0500
Subject: [PATCH] Use the job owner rather than the uid for credential naming

The job owner will be more unique than the logged in username
of on the condor starter.  For example, if the worker node runs
all jobs under the nobody user, then all credentials will be mangled.
---
 src/condor_starter.V6.1/jic_shadow.cpp | 20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/src/condor_starter.V6.1/jic_shadow.cpp b/src/condor_starter.V6.1/jic_shadow.cpp
index 805b105..72fa75e 100644
--- a/src/condor_starter.V6.1/jic_shadow.cpp
+++ b/src/condor_starter.V6.1/jic_shadow.cpp
@@ -2574,8 +2574,14 @@ JICShadow::initUserCredentials() {
 	}
 
 
-	// get username
-	MyString user = get_user_loginname();
+	// Use Job Owner rather than login name
+	// REALLY HACKY, should us something more specific, like using the UID Domain or something
+	std::string user;
+	if ( ! job_ad->EvaluateAttrString("Owner", user) ) {
+		dprintf( D_FAILURE, "JICShadow::initUserCredentials(): "
+				 "Unable to get Owner attribute from jobid\n");
+		return false;
+	}
 	MyString domain = "DOMAIN";
 
 	// remove mark on update for "mark and sweep"
@@ -2719,8 +2725,14 @@ JICShadow::refreshSandboxCredentials()
 	char  *ccbuf = 0;
 	size_t cclen = 0;
 
-	// get username
-	MyString user = get_user_loginname();
+	// Use Job Owner rather than login name
+	// REALLY HACKY, should us something more specific, like using the UID Domain or something
+	std::string user;
+	if ( ! job_ad->EvaluateAttrString("Owner", user) ) {
+		dprintf( D_FAILURE, "JICShadow::refreshSandboxCredentials(): "
+				 "Unable to get Owner attribute from jobid\n");
+		return false;
+	}
 
 	// declaring at top since we use goto for error handling
 	priv_state priv;
-- 
2.10.1

